Extend datamap logic with alternative worksheet implementation

Freymaurer opened the pull request:
Following the current ARC-specification each Assay/Study can be related to at most **one** `isa.datamap.xlsx` file. And each `isa.datamap.xlsx` fiel MUST contain at most most **one** `isa_datamap` worksheet, which MUST contain at most **one** table named `datamapTable`.

This allows us to give an alternative implementation as additional named worksheet directly in assay/study. Allowing this structure would make development for Swate easier, as well as maintaining familiar behavior of downloading only one file.

Should one assay have a `isa_datamap` sheet with a `datamapTable`, as well as a `isa.datamap.xlsx` at the correct position, the question arises how to handle this "double information. In my opinion there are several options on how to handle this:

1. Extend `isa.datamap.xlsx` information with sheet (KEEP BOTH)
2. Prioritize `isa.datamap.xlsx`
3. Prioritize sheet

The current PR defines variant 3. I think, that this is the least "magic" by also choosing "specific" over "general".

Let me know what you think @HLWeil @kappe-c @floWetzels @ZimmerD @kMutagene @chgarth @muehlhaus @gdoniparthi

HLWeil reviewed (changes requested):
Seems like a sensible thing to do.

Two things though:

1. Could you remote the `Datamap Sheet` section you added and instead mention in the `Assay file` and `Study file` sections?
2. Could you also add a small mention with basically the same information in the [Study](https://github.com/nfdi4plants/ARC-specification/blob/main/ARC%20specification.md#study-and-resources) and [Assay](https://github.com/nfdi4plants/ARC-specification/blob/main/ARC%20specification.md#assay-data-and-metadata) sections of the main ARC-specification file? Best place would be the paragraphs where there's talk about the datamap file.

Freymaurer commented:
I started working on your request, but noticed that this would require a lot of information duplication. Stating the same information at multiple points in the specification. Therefore i propose another solution.

- I added references to the ARC-specification
- In ISA-XLSX specification i added references in study and assay to datamap sheet.
- Both datamap file and datamap sheet do now crossreference to datamap table which contains specific column information

HLWeil commented on ISA-XLSX.md:
```diff
@@ -133,18 +135,24 @@ Additionally, the `Assay File` SHOULD contain one or more [`Annotation Table she
 
 The `Datamap File` acts as an extension of the `data` nodes defined in the [`Study and Assay graphs section`](https://isa-specs.readthedocs.io/en/latest/isamodel.html#study-and-assay-graphs) from the ISA Abstract Model.
 
+## Datamap Sheet
```
Remove this section

HLWeil commented on ISA-XLSX.md:
```diff
@@ -118,6 +118,8 @@ Additionally, the `Study File` SHOULD contain one or more [`Annotation Table she
 
 Therefore, the main entities of the `Study File` should be `Sources` and `Samples`.
 
+Any Study MAY contain datamap references as described in the [`Datamap Sheet`](#datamap-sheet) section.
```
Keep this

HLWeil replied:
The links need to be fixed (#datamap-table-sheet)

HLWeil commented on ISA-XLSX.md:
```diff
@@ -133,18 +135,24 @@ Additionally, the `Assay File` SHOULD contain one or more [`Annotation Table she
 
 Therefore, the main entities of the `Assay File` should be `Samples` and `Data`.
 
+Any Assay MAY contain datamap references as described in the [`Datamap Sheet`](#datamap-sheet) section.
```
Keep this

HLWeil commented on ISA-XLSX.md:
```diff
@@ -133,18 +135,24 @@ Additionally, the `Assay File` SHOULD contain one or more [`Annotation Table she
 
 The `Datamap` represents a set of explanations about the `data` entities defined in `assays` and `studies`.
 
-The `Datamap File` MUST contain one [`Datamap table sheet`](#datamap-table-sheets). This sheet MUST be named `isa_datamap`.
```
Undo

HLWeil commented on ISA-XLSX.md:
```diff
@@ -785,19 +793,19 @@ If we pool two sources into a single sample, we might represent this as:
 | source1       | sample collection | sample1       |
 | source2       | sample collection | sample1       |
 
-# Datamap Table sheets
```
Undo

HLWeil commented on ISA-XLSX.md:
```diff
@@ -785,19 +793,19 @@ If we pool two sources into a single sample, we might represent this as:
-# Datamap Table sheets
+# Datamap Table
 
-`Datamap Table sheets` are used to describe the contents of data files. 
```
Undo

HLWeil commented on ISA-XLSX.md:
```diff
@@ -785,19 +793,19 @@ If we pool two sources into a single sample, we might represent this as:
-`Datamap Table sheets` are used to describe the contents of data files. 
+`Datamap Table` are used to describe the contents of data files. 
 
-In the `Datamap Table sheets`, column headers MUST have the first letter of each word in upper case, with the exception of the referencing label (REF).
```
Undo

HLWeil commented on ISA-XLSX.md:
```diff
@@ -785,19 +793,19 @@ If we pool two sources into a single sample, we might represent this as:
 
 The content of the datamap table MUST be placed in an `xlsx table` whose name equals `datamapTable`. Each sheet MUST contain at most one such datamap table. Only cells inside this table are considered as part of the formatted metadata.
 
-`Datamap Table sheets` are structured with fields organized on a per-row basis. The first row MUST be used for column headers. Each body row is an implementation of a `data` node.
```
Undo

HLWeil commented on ISA-XLSX.md:
```diff
@@ -785,19 +793,19 @@ If we pool two sources into a single sample, we might represent this as:
 
 ## Data column
 
-Every `Datamap Table sheet` MUST contain a `Data` column. Every  object in this column MUST correspond to a relevant data resource location, following the [Data Path Annotation](/ARC%20specification.md#data-path-annotation) patterns. If the annotation of the `Data` node refers not to the complete resource, but a part of it, a `Selector` MAY be added. This Selector MUST be separated from the resou ...
```
Undo

HLWeil commented on ARC specification.md:
```diff
@@ -76,9 +76,9 @@ ARCs are based on a strict separation of data and metadata content into study ma
 Each ARC is a directory containing the following elements:
 
-- *Studies* are collections of material and resources used within the investigation. Study-level metadata is stored in [ISA-XLSX](#isa-xlsx-format) format in a `isa.study.xlsx` file, which MUST exist to specify the input material or data resources. Resources MAY include biological materials (e.g. plant samples, analytical standards) created during the current investigation. Resources MAY further  ...
+- *Studies* are collections of material and resources used within the investigation. Study-level metadata is stored in [ISA-XLSX](#isa-xlsx-format) format in a `isa.study.xlsx` file, which MUST exist to specify the input material or data resources. Resources MAY include biological materials (e.g. plant samples, analytical standards) created during the current investigation. Resources MAY further  ...
```
format in a `isa.datamap.xlsx` file **in the study folder** or `isa_datamap` worksheet in the **isa.study.xlsx** file

HLWeil commented on ARC specification.md:
```diff
@@ -76,9 +76,9 @@ ARCs are based on a strict separation of data and metadata content into study ma
+- *Studies* are collections of material and resources used within the investigation. Study-level metadata is stored in [ISA-XLSX](#isa-xlsx-format) format in a `isa.study.xlsx` file, which MUST exist to specify the input material or data resources. Resources MAY include biological materials (e.g. plant samples, analytical standards) created during the current investigation. Resources MAY further  ...
 
-- *Assays* correspond to outcomes of experimental assays or analytical measurements (in the interpretation of the ISA model) and are treated as immutable data. Each assay is a collection of files, together with a corresponding metadata file, stored in a subdirectory of the top-level subdirectory `assays`. Assay-level metadata is stored in [ISA-XLSX](#isa-xlsx-format) format in a `isa.assay.xlsx`  ...
+- *Assays* correspond to outcomes of experimental assays or analytical measurements (in the interpretation of the ISA model) and are treated as immutable data. Each assay is a collection of files, together with a corresponding metadata file, stored in a subdirectory of the top-level subdirectory `assays`. Assay-level metadata is stored in [ISA-XLSX](#isa-xlsx-format) format in a `isa.assay.xlsx`  ...
```
format in a `isa.datamap.xlsx` file **in the assay folder** or `isa_datamap` worksheet in the **isa.assay.xlsx** file

HLWeil commented on ISA-XLSX.md:
```diff
@@ -26,7 +26,7 @@ This document describes the ISA Abstract Model reference implementation specifie
   - [Comments](#comments)
   - [Examples](#examples-1)
-- [Datamap Table sheets](#datamap-table-sheets)
+- [Datamap Table](#datamap-table)
```
Undo

HLWeil reviewed (changes requested):
I see the problem of duplication, but now some inconsistencies were introduced into the isa-xlsx specification. Especially splitting the `datamap table sheet` into a `table` and `sheet` section. This is inconsistent with the `annotation table sheet` and `top level metadata sheet` sections. 

Therefore I propose to undo this split, mention the possibility to have the  `datamap table sheet` directly in the assay or study file (as you did) and be a little bit more specific in the assay and study sections of the arc-specification.

Freymaurer commented:
Done! Have a look!

HLWeil commented on ISA-XLSX.md:
```diff
@@ -133,6 +135,8 @@ Additionally, the `Assay File` SHOULD contain one or more [`Annotation Table she
 
 Therefore, the main entities of the `Assay File` should be `Samples` and `Data`.
 
+Any Assay MAY contain datamap references as described in the [`Datamap Sheet`](#datamap-sheet) section.
```
The links need to be fixed (#datamap-table-sheet)

HLWeil reviewed (approved):
Thanks for the adjustments. Just saw that the links need to be fixed. Else it's ready to go.

Freymaurer commented:
@HLWeil fixed! 👍
